src/all.js

```
@@ -2,28 +2,42 @@ const { parseFileRedirects } = require('./line_parser')
 const { mergeRedirects } = require('./merge')
 const { parseConfigRedirects } = require('./netlify_config_parser')
 const { normalizeRedirects } = require('./normalize')
+const { splitResults, concatResults } = require('./results')
 
 // Parse all redirects from `netlify.toml` and `_redirects` file, then normalize
 // and validate those.
 const parseAllRedirects = async function ({ redirectsFiles = [], netlifyConfigPath, ...opts } = {}) {
-  const [fileRedirects, configRedirects] = await Promise.all([
-    getFileRedirects(redirectsFiles),
-    getConfigRedirects(netlifyConfigPath),
-  ])
-  const normalizedFileRedirects = normalizeRedirects(fileRedirects, opts)
-  const normalizedConfigRedirects = normalizeRedirects(configRedirects, opts)
-  return mergeRedirects({ fileRedirects: normalizedFileRedirects, configRedirects: normalizedConfigRedirects })
+  const [
+    { redirects: fileRedirects, errors: fileParseErrors },
+    { redirects: configRedirects, errors: configParseErrors },
+  ] = await Promise.all([getFileRedirects(redirectsFiles), getConfigRedirects(netlifyConfigPath)])
+  const { redirects: normalizedFileRedirects, errors: fileNormalizeErrors } = normalizeRedirects(fileRedirects, opts)
+  const { redirects: normalizedConfigRedirects, errors: configNormalizeErrors } = normalizeRedirects(
+    configRedirects,
+    opts,
+  )
+  const { redirects, errors: mergeErrors } = mergeRedirects({
+    fileRedirects: normalizedFileRedirects,
+    configRedirects: normalizedConfigRedirects,
+  })
+  const errors = [
+    ...fileParseErrors,
+    ...fileNormalizeErrors,
+    ...configParseErrors,
+    ...configNormalizeErrors,
+    ...mergeErrors,
+  ]
+  return { redirects, errors }
```

Any reason for the `mergeErrors` to not contain the concatenated result of all of the errors showing below? 🤔

---

Ah, I guess we can have errors within the initial normalisation steps that we'll need to return the errors for correct?

---

Yes, exactly.
The `mergeErrors` are only the errors happening during the merging step (actually at the moment, there are no such errors at that step), while the others are errors happening during the parsing and normalization steps for `netlify.toml` and `_redirects`, respectively. We return all of them as a single array, and filter out any redirects which produced an error.